doc/reference/trd-appid.md

```diff
@@ -0,0 +1,523 @@
+binary. It allows the Tock kernel to know about the provenance and
+origin of the binary and make access control or security decisions
+based on this information. For example, a kernel may allow only
+applications whose credentials use a particular trusted public key to
+access restricted functionality, but restrict other applications to
+use a subset of available system calls.
+
+Application identifiers are distinct from process
+identifiers. An application identifier is per-application (persists
+across restarts of a process binary, for example), while a process
+identifier identifies a particular execution of that binary. At any
+time on a Tock device, each process has a unique process identifier,
+but they can be re-used over time (like POSIX process identifiers).
+
+As the above examples illustrate, application credentials can vary in
+size and content. The credentials that a kernel's verifier policy will
+accept depends on its use case. Certain devices will only accept
+credentials which include a particular public key, while others will
+accept many. Furthermore, the internal format of these credentials can
+vary.  Finally, the cryptography used in credentials can vary, either
+due to security policies or certification requirements.
+
+Because the verifier policy is responsible for assigning application
+identifiers to process binaries, it is possible for the same process
+binary to have different application identifiers on different Tock
+systems.  For example, suppose a process binary has two application
+credential TBF headers: one signs with a key A, and the other with key
+B. Tock systems using a verifier policy that accepts key A may assign
+A as the global application identifier, while Tock systems using a
+different verifier policy that accepts key B may assign B as the
+global application identifier.
+
+4 Credentials in Tock Binary Format Headers
+===============================
+
+To support credentials in Tock binaries, the Tock Binary Format has
+a `TbfHeaderV2Credentials` TLV. This TLV is variable length
+and has two fields:
+
+```rust
+pub struct TbfHeaderV2Credentials {
+    format: TbfHeaderV2CredentialsType,
+    data: &[u8],
+}
+```
+
+The `TbfHeaderV2CredentialsType` defines the format and size of `data`
+field. A `TbfHeaderV2CredentialsType` value MUST have a fixed
+data size and format. Currently supported values are:
+
+```rust
+pub enum TbfHeaderV2CredentialsType {
+    Padding = 0,
+    CleartextID = 1,
+    Rsa3072Key = 2,
+    Rsa4096Key = 3,
+    Rsa3072KeyWithID = 4,
+    Rsa4096KeyWithID = 5,
+}
+```
+
+**These are not intended to be final or prescriptive. They are merely some examples
+of what kind of information we might put here. Among other things, the exact format
+of the data blocks needs to be more precise. -pal**
+
+The `Padding` value has a variable length. It has a 32-bit field
+specifying its total length. This credentials type is used to
+reserve space for future credentials or pad their placement.
```

Comment:
A process binary is 32kB in size (e.g., needs power-of-2 sizing as in CortexM). The application binary is 21kB. The footers are 2kB. A padding footer would be placed in the last 8kB.

Currently, there's no notion of explicit padding. But since we now parse what's at the end, we need a way to fill it without accidentally parsing empty space.